fetchconfig/fileOps.cpp

```diff
@@ -0,0 +1,35 @@
+/*
+ * fileOps.cpp
+ *
+ *      Author: Zac
+ *     Contact: codeohms@protonmail.com
+ *  Created on: 17 Sep. 2018
+ */
+
+
+#include "fileOps.h"
+
+
+int getUserConfigDir(std::string& dir)
+{
+#ifndef  WIN32
+    char* path = getenv("HOME");
+    if(path == NULL)
+        return 1;
+    
+#else
+    static char path[MAX_PATH];

```

Comment:
This won't work for users with non-ascii chars in username. Look for SHGetKnownFolderPath in gimx.c. I'll try to move all file-specific code into a shared library.

Reply:
I moved the file-specific code in a new library, and pushed the code into the gimxfile branch. Rebase your branch on the gimxfile branch, and use gfile_homedir instead of getUserConfigDir. Returned string is statically allocated, so don't forget to free it.